pkg/clusterversion/setting.go

```diff
@@ -165,7 +165,7 @@ func (cv *clusterVersionSetting) ValidateVersionUpgrade(
 	_ context.Context, sv *settings.Values, curRawProto, newRawProto []byte,
 ) error {
 	var newCV ClusterVersion
-	if err := protoutil.Unmarshal(newRawProto, &newCV); err != nil {
```

Comment:
We don't strictly need to change all of these, they're not hot paths. I think `activeVersionOrEmpty`, and possibly `Decode`, are the ones we care about.

Reply:
+1

Honestly we could also just move clusterversion.proto into another package that pkg/settings is allowed to depend on and then store a *ClusterVersion in the atomic instead of a []byte so there'd be no unmarshal -- allocating or otherwise -- on the hot path. But that's a bigger change, so if you're eyeing a back port to resolve an observed customer issue, this more targeted removal of just the alloc, and just on the hot path, probably makes sense for now.

Reply:
Leaving as is so that this file consistently uses one way. LMK if feel strongly.
I will file an issue to get rid of protos in this package.